src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -41,201 +39,182 @@
+
+    /**
+     * The hours the colony is without contact with its players.
+     */
+    private int lastContactInHours = 0;
+
+    /**
+     * Whether or not this colony may be auto-deleted.
+     */
+    private boolean canColonyBeAutoDeleted = true;
+
+    /**
+     * Variable to determine if its currently day or night.
+     */
+    private boolean isDay = true;
+
+    /**
+     * The world the colony currently runs on.
+     */
     @Nullable
-    private       World                           world             = null;
-    //  Updates and Subscriptions
+    private World world = null;
+
+    /**
+     * List of players subscribing to the colony.
+     */
     @NotNull
-    private       Set<EntityPlayerMP>             subscribers      = new HashSet<>();
-    private       boolean                         isDirty          = false;
-    private       boolean                         isCitizensDirty  = false;
-    private       boolean                         isBuildingsDirty = false;
-    private       boolean                         manualHiring     = false;
-    private       boolean                         manualHousing     = false;
-
-    private       boolean                         isFieldsDirty    = false;
-    private       String                          name             = "ERROR(Wasn't placed by player)";
-    private BlockPos         center;
-    //  Administration/permissions
+    private Set<EntityPlayerMP> subscribers = new HashSet<>();
+
+    /**
+     * Variables taking care of updating the views.
+     */
+    private boolean isDirty          = false;
+
+    /**
+     * The hiring mode in the colony.
+     */
+    private boolean manualHiring = false;
+
+    /**
+     * The housing mode in the colony.
+     */
+    private boolean manualHousing = false;
+
+    /**
+     * The name of the colony.
+     */
+    private String name = "ERROR(Wasn't placed by player)";
+
+    /**
+     * The center of the colony.
+     */
+    private BlockPos center;
+
+    /**
+     * The colony permission object.
+     */
     @NotNull
-    private Permissions      permissions;
-    @Nullable
-    private BuildingTownHall townHall;
-    private int topCitizenId = 0;
-    private int maxCitizens  = Configurations.gameplay.maxCitizens;
+    private Permissions permissions;
 
-    private double overallHappiness = 5;
+    /**
+     * Overall happyness of the colony.
+     */
+    private double overallHappiness = DEFAULT_OVERALL_HAPPYNESS;
 
     /**
      * Amount of ticks passed.
      */
     private int ticksPassed = 0;
 
+    /**
+     * The request manager assigned to the colony.
+     */
+    private IRequestManager requestManager;
+
     /**
      * Constructor for a newly created Colony.
      *
      * @param id The id of the colony to create.
      * @param w  The world the colony exists in.
      * @param c  The center of the colony (location of Town Hall).
      */
+    @SuppressWarnings("squid:S2637")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2637' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)